### PR TITLE
add a quickstart for retention

### PR DESCRIPTION
**Pull request by y-f-u**
retention quick start.

Part of https://github.com/spiceai/spiceai/issues/1088

### CODE REVIEW

File: retention/README.md
Comment (phillipleblanc) [inline retention/README.md:1]: Please add a link from the top-level `README.md` to this quickstart

File: README.md [hunk 1]
```diff
@@ -33,3 +33,7 @@ Designed to take 1-5 mins each to get started quickly!
 ## Performance
 
 - [TPC-H Benchmarking](./tpc-h/README.md)
+
+## Other features
+
+- [Retention](./retention/README.md)
```
Comment (lukekim) [inline README.md]: Needs to be more descriptive. Something like "Data Retention Policies"?

File: retention/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# Quickstart for retention policies on accelerated dataset
```
Comment (lukekim) [inline retention/README.md]:
```suggestion
# Accelerated Dataset Retention Policy Quickstart
```

File: retention/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+**Step 1.** Initialize and start Spice
+
+```bash
+spice init retention-quickstart
+cd retention-quickstart
+```
+
+**Step 2.** Add a dataset with retention policies by editing spicepod.yaml
```
Comment (lukekim) [inline retention/README.md]:
```suggestion
**Step 2.** Add a dataset with a retention policy by editing spicepod.yaml
```

File: retention/README.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+datasets:
+  - from: s3://spiceai-demo-datasets/taxi_trips/2024/
+    name: taxi_trips
+    time_column: tpep_pickup_datetime
+    acceleration:
+      retention_enabled: true
+      enabled: true
+      refresh_interval: 10m
+      retention_check_interval: 60s
+      retention_period: 35040h # 4 years, this will eliminate 5 rows of bad data in the public dataset
```
Comment (lukekim) [inline retention/README.md]:
```suggestion
      retention_period: 35040h # 4 years, this will evict 5 rows of data from the dataset
```

**Review by lukekim — DISMISSED** [review]

**Review by ewgenius — APPROVED** [review]

**Pull request merged**